packages/react-sdk/src/state/synchronizedDocumentImpl.test.ts

```diff
@@ -707,6 +707,94 @@ describe('SynchronizedDocumentImpl', () => {
     expect(await statistics).toEqual([]);
     expect(await loading).toEqual([]);
   });
+
+  describe('persist', () => {
+    it('should create a snapshot when there are changes', async () => {
+      const doc = createExampleDocument();
+      const store = createStore({ widgetApi });
+
+      const synchronizedDocument = new SynchronizedDocumentImpl(
+        doc,
+        store,
+        communicationChannel,
+        storage,
+        '$document-0',
+      );
+
+      const outstandingStatistics = firstValueFrom(
+        synchronizedDocument
+          .observeDocumentStatistics()
+          .pipe(take(2), toArray()),
+      );
+      const statistics = firstValueFrom(
+        synchronizedDocument.observeDocumentStatistics().pipe(skip(2)),
+      );
+
+      const isReady = firstValueFrom(
+        synchronizedDocument
+          .observeIsLoading()
+          .pipe(filter((loading) => !loading)),
+      );
+
+      doc.performChange((doc) => doc.set('num', 10));
+
+      synchronizedDocument.persist();
+
+      expect(await outstandingStatistics).toEqual([
+        {
+          contentSizeInBytes: 20,
+          documentSizeInBytes: expect.any(Number),
+          snapshotOutstanding: false,
+          snapshotsReceived: 0,
+          snapshotsSend: 0,
+        },
+        {
+          contentSizeInBytes: 20,
+          documentSizeInBytes: expect.any(Number),
+          snapshotOutstanding: true,
+          snapshotsReceived: 0,
+          snapshotsSend: 0,
+        },
+      ]);
+
+      await isReady;
+
+      await waitFor(() => {
+        expect(widgetApi.sendRoomEvent).toHaveBeenCalledTimes(2);
```

Comment:
Feels awkward to check for generic sending of any type of room event when the test is about snapshot events.

I realize that other previous tests in this suite had this same approach but I also see that we can actually check for event types if we use `toHaveBeenCalledWith`.

But since we care about the number of calls, maybe complement it with `toHaveBeenLastCalledWith`?

Anyway, not a blocker, just thinking if / how it can be improved.

Reply:
Updated, it now checks for snapshot events